enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java

```diff
@@ -141,14 +143,34 @@ private void addServices(KubernetesListBuilder builder) {
         }
     }
 
+    private String getAppName() {
+        try {
+            if (getContext().getProjectClassLoaders().isClassInCompileClasspath(true)) {
+                Properties properties = SpringBootUtil.getSpringBootApplicationProperties(getContext().getProjectClassLoaders().getCompileClassLoader());
+                return properties.getProperty("spring.application.name");
+            }
+        } catch (Exception ex) {
+            log.error("Error while reading the spring-boot configuration", ex);
+        }
+        return null;
+    }
+
+    private String getServiceName() {
+        if (getAppName() != null) {
```

Comment:
Can you skip calling `getAppName` twice? Do something like:

```
String appName = getAppName();
if (appName!=null) {
return appName;
} else {
....
}
```

After that, I think we are ready to merge.